test/widgets/register_page_test.dart

```diff
@@ -0,0 +1,322 @@
+import 'package:motis_mitfahr_app/util/buttons/loading_button.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:motis_mitfahr_app/welcome/pages/after_registration_page.dart';
+import 'package:motis_mitfahr_app/welcome/pages/register_page.dart';
+import 'package:progress_state_button/progress_button.dart';
+
+import '../util/mocks/mock_server.dart';
+import '../util/mocks/request_processor.dart';
+import '../util/mocks/request_processor.mocks.dart';
+import '../util/pump_material.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  const String username = 'motismitfahrapp';
+  const String email = 'motismitfahrapp@gmail.com';
+  const String password = 'abc&EFG0';
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+    SupabaseManager.setCurrentProfile(null);
+  });
+
+  group('RegisterPage', () {
+    final Finder usernameFieldFinder = find.byKey(const Key('registerUsernameField'));
+    final Finder emailFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerEmailField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerPasswordField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordConfirmFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerPasswordConfirmField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder registerButtonFinder = find.byType(LoadingButton);
+
+    testWidgets('Shows the register page', (WidgetTester tester) async {
+      await pumpMaterial(tester, const RegisterPage());
+
+      final Finder registerFormFinder = find.byType(RegisterForm);
+      final RegisterFormState formState = tester.state(registerFormFinder);
+
+      expect(formState.buttonState, ButtonState.idle);
+    });
+
+    testWidgets('Validates the email', (WidgetTester tester) async {
+      await pumpMaterial(tester, const RegisterPage());
+
+      final RegisterFormState formState = tester.state(find.byType(RegisterForm));
+
+      final FormFieldState emailField = tester.state(emailFieldFinder);
+
+      // Not validated yet, so no error
+      expect(emailField.hasError, isFalse);
+
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Empty, so error
+      expect(emailField.hasError, isTrue);
+
+      await tester.enterText(emailFieldFinder, '123noemail');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Not real email, so error
+      expect(emailField.hasError, isTrue);
+
+      await tester.enterText(emailFieldFinder, email);
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Real email, so no error
+      expect(emailField.hasError, isFalse);
+    });
+
+    testWidgets('Validates the username', (WidgetTester tester) async {
```

Comment:
please add a case where the username is to long.

Reply:
Ah actually we forgot to add the restriction in the register field. I will add and test it.